src/typedefs.jl

```diff
@@ -7,32 +7,54 @@ const DefaultRefType = UInt32
 # * `R` integer type for referencing category levels
 # * `V` categorical value type
 mutable struct CategoricalPool{T, R <: Integer, V}
-    index::Vector{T}        # category levels ordered by their reference codes
+    levels::Vector{T}       # category levels ordered by their reference codes
     invindex::Dict{T, R}    # map from category levels to their reference codes
-    order::Vector{R}        # 1-to-1 map from `index` to `level` (position of i-th category in `levels`)
-    levels::Vector{T}       # category levels ordered by externally specified order
     valindex::Vector{V}     # "category value" objects 1-to-1 matching `index`
```

Comment:
What's the purpose of `valindex`? Is the benefit just that it makes the CategoricalValue objects once instead of repeatedly on-demand?

Reply:
Yeah. I wanted to try removing it and see whether it makes a difference for performance. It might actually be faster in some cases as the compiler may be able to detect that it only needs to use the reference code and not load the pointer to the pool at all. But that's for another PR.